src/modules/m_sqloper.cpp

```diff
@@ -76,9 +76,21 @@ class OpMeQuery : public SQLQuery
 		}
 	}
 
-	bool OperUser(User* user, const std::string &pattern, const std::string &type)
+	bool OperUser(User* user, const std::string &pattern, const std::string &type, const std::string &hashedpass)
 	{
 		ServerConfig::OperIndex::const_iterator iter = ServerInstance->Config->OperTypes.find(type);
+		HashProvider* hash = ServerInstance->Modules->FindDataService<HashProvider>("hash/" + hashtype);
+		
+		if(hash)
+		{
+			if(!hash->Compare(password,hashedpass))
+		    		return false;
+		}
+		else
+		{
+			if(hashedpass.compare(password) != 0)
+				return false;
+		}
```

Comment:
I think I've done that because if the HashProvider is not found, I assume that hashedpass is in fact the plaintext password.
One way to do it, is to create a NoHash hash algorithm.